From 79fbef9d323a6ce4d2c765ebf33e43eabc5cf7e3 Mon Sep 17 00:00:00 2001
From: Juliusz Sosinowicz <juliusz@wolfssl.com>
Date: Wed, 3 Apr 2024 17:23:45 +0200
Subject: [PATCH 24/24] wolfSSL: test_ap_wpa2_eap_fast_server_oom

wolfSSL sends the session ticket extension on the first TLS handshake. This causes the malloc error to be triggered on the first connection instead of the second and the failure event is not registered in the python test.

Signed-off-by: Juliusz Sosinowicz <juliusz@wolfssl.com>
---
 tests/hwsim/test_ap_eap.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/hwsim/test_ap_eap.py b/tests/hwsim/test_ap_eap.py
index 580660e592..9b1ca119b7 100644
--- a/tests/hwsim/test_ap_eap.py
+++ b/tests/hwsim/test_ap_eap.py
@@ -4277,6 +4277,9 @@ def test_ap_wpa2_eap_fast_prf_oom(dev, apdev):
 def test_ap_wpa2_eap_fast_server_oom(dev, apdev):
     """EAP-FAST/MSCHAPv2 and server OOM"""
     check_eap_capa(dev[0], "FAST")
+    tls = dev[0].request("GET tls_library")
+    if not tls.startswith("OpenSSL"):
+        raise HwsimSkip("TLS library is not OpenSSL: " + tls)
 
     params = int_eap_server_params()
     params['dh_file'] = 'auth_serv/dh.conf'
-- 
2.34.1

